server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -549,15 +615,86 @@ public void flushOneMemTable() {
     memTableToFlush = flushingMemTables.getFirst();
     if (logger.isInfoEnabled()) {
       logger.info("{}: {} starts to flush a memtable in a flush thread", storageGroupName,
-              tsFileResource.getFile().getName());
+          tsFileResource.getFile().getName());
     }
     // signal memtable only may appear when calling asyncClose()
     if (!memTableToFlush.isSignalMemTable()) {
-      MemTableFlushTask flushTask = new MemTableFlushTask(memTableToFlush, writer,
-          storageGroupName);
       try {
+        boolean isVm = false;
+        boolean isFull = false;
+        MemTableFlushTask flushTask;
+        if (config.isEnableVm()) {
+          long vmPointNum = 0;
+          for (RestorableTsFileIOWriter vmWriter : vmWriters) {
+            Map<String, Map<String, List<ChunkMetadata>>> metadatasForQuery = vmWriter
+                .getMetadatasForQuery();
+            for (String device : metadatasForQuery.keySet()) {
+              Map<String, List<ChunkMetadata>> chunkMetadataListMap = metadatasForQuery.get(device);
+              for (String sensor : chunkMetadataListMap.keySet()) {
+                for (ChunkMetadata chunkMetadata : chunkMetadataListMap.get(sensor)) {
+                  vmPointNum += chunkMetadata.getNumOfPoints();
+                }
+              }
+            }
+          }
+          // all flush to target file
+          if ((
+              (vmPointNum + memTableToFlush.getTotalPointsNum()) / memTableToFlush.getSeriesNumber()
```

Comment:
The denominator should add the number of series that exist in vmWriters but not in memTableToFlush